ananse/peakpredictor.py

```diff
@@ -0,0 +1,617 @@
+        if self._atac_data is not None:
+            tmp = tmp.join(self._atac_data)
+        if self._histone_data is not None:
+            tmp = tmp.join(self._histone_data)
+
+        if self.region_type == "reference":
+            tmp = tmp.join(self._avg)
+            tmp = tmp.join(self._dist)
+        tmp = tmp.dropna()
+        # logger.debug(str(self._X_columns))
+        return tmp[self._X_columns]
+
+    def _load_model(self, factor):
+        model = None
+        if factor in self.factor_models:
+            logger.info(f"Using {factor} model")
+            model = self.factor_models[factor]
+        elif factor in self.motif_graph:
+            paths = {
+                p: v
+                for p, v in nx.single_source_dijkstra_path_length(
+                    self.motif_graph, factor
+                ).items()
+                if p in self.factor_models
+            }
+            try:
+                sub_factor = list(paths.keys())[0]
+                logger.info(f"Using {factor} motif with {sub_factor} model weights")
+                model = self.factor_models[sub_factor]
+                # factor = sub_factor
+            except Exception:
+                logger.info(f"No match for {factor} based on motifs")
+        if model is None:
+            logger.info(f"No related TF found for {factor}, using general model")
+            model = self.factor_models["general"]
+
+        return model, factor
+
+
+def _check_input_regions(regionfiles, genome, outdir=".", verbose=True, force=False):
+    # Load regions from BED or region text file
+    if regionfiles is None:
+        # Keep regions to None, use reference regions.
+        return
+
+    infile = regionfiles[0]
+    if len(regionfiles) > 1:
+        # merge files
+        peak_width = 200
+        cbed = CombineBedFiles(genome=genome, peakfiles=regionfiles, verbose=verbose)
+        combined_bed = os.path.join(outdir, "regions_combined.bed")
+        cbed.run(outfile=combined_bed, width=peak_width, force=force)
+        infile = combined_bed
+
+    df = pd.read_table(infile, header=None)
+    if df.shape[1] >= 3:
+        regions = (
+            df.iloc[:, 0].astype(
+                str
+            )  # For Ensembl genome names, make sure it's a string
+            + ":"
+            + df.iloc[:, 1].astype(str)
+            + "-"
+            + df.iloc[:, 2].astype(str)
+        ).tolist()
+    else:
+        regions = df.iloc[:, 0].tolist()
+    return regions
+
+
+def _check_input_files(*args):
+    files = []
+    for arg in args:
+        if arg is None:
+            continue
+        if isinstance(arg, list):
+            files.extend(arg)
+        else:
+            files.append(arg)
+
+    all_files_found = True
+    for fname in files:
+        if not os.path.exists(fname):
+            logger.exception(f"Could not find {fname}!")
+            all_files_found = False
+
+    if not all_files_found:
+        exit(1)
+
+
+def predict_peaks(
```

Comment:
this is a point on formality (and can therefore be ignored if it takes too long).

half of this function is checking the user input, the other half is running the code. The latter part definitely belongs here, but the input checking should maybe move to `commands/`, as it is the control script.

Reply:
I see your point. However, I like that the code in commands is actually very minimal. This means that the exact same functionality will be available as API-style functionality. What can be done is to move it to a separate function.